proxy/src/main/java/net/md_5/bungee/ServerConnection.java

```diff
@@ -53,6 +59,8 @@ public void sendPacketQueued(DefinedPacket packet)
             Protocol encodeProtocol = ch.getEncodeProtocol();
             if ( !encodeProtocol.TO_SERVER.hasPacket( packet.getClass(), ch.getEncodeVersion() ) )
             {
+                // we should limit this so bad api usage won't oom the server.
+                Preconditions.checkState( packetQueue.size() <= 2 << 11, "too many queued packets" );
```

Comment:
Just use 4096, clearer for small numbers

Reply:
okay